react/src/components/Table/ColumnVisibilityModal.tsx

```diff
@@ -1,70 +1,152 @@
+                }}
+                helperText="Only columns within the same group can be reordered."
             >
                 {headerGroups[0].headers.map((g, id) => (
-                    <div key={id}>
-                        <Checkbox
-                            label={g.Header as string}
-                            checked={g.isVisible}
-                            onClick={() => toggleGroupVisibility(g)}
-                        />
-                        {g.columns?.map(
-                            (c, id) =>
-                                c.type !== 'fixed' &&
-                                c.type !== 'empty' && (
-                                    <div key={id} style={{ paddingLeft: 20 }}>
-                                        <Checkbox
-                                            label={c.Header as string}
-                                            checked={c.isVisible}
-                                            onClick={() => {
-                                                if (
-                                                    c.parent &&
-                                                    g.columns?.filter(c => c.isVisible).length === 1
-                                                ) {
-                                                    toggleHideColumn(c.id, c.isVisible);
-                                                    toggleHideColumn(
-                                                        camelize(`empty ${c.parent.id}`),
-                                                        !c.isVisible
-                                                    );
-                                                } else {
-                                                    toggleHideColumn(c.id, c.isVisible);
-                                                }
-                                            }}
-                                        />
-                                    </div>
-                                )
-                        )}
-                    </div>
+                    <DragDropContext key={id} onDragEnd={onDragEnd}>
+                        <Droppable key={id} droppableId={JSON.stringify(id)}>
+                            {(droppableProvided, snapshot) => (
+                                <div
+                                    ref={droppableProvided.innerRef}
+                                    {...droppableProvided.droppableProps}
+                                >
+                                    <Checkbox
+                                        label={g.Header as string}
+                                        checked={g.isVisible}
+                                        onClick={() => toggleGroupVisibility(g)}
+                                    />
+                                    {order.map(
+                                        (c, id) =>
+                                            g.columns?.find(column => column.id === c.id) &&
```

Comment:
The code here is getting pretty hard to read. Note to self to open another issue to refactor react-table

Reply:
Sure, you can also add on to this issue opened earlier: [https://github.com/ccmbioinfo/osmp/issues/216](https://github.com/ccmbioinfo/osmp/issues/216)